### PR TITLE
Simplify Buttons Props

### PR DESCRIPTION
**Pull request by djhi**

### CODE REVIEW

**Review by slax57 — APPROVED** [review]

File: packages/ra-ui-materialui/src/button/Button.tsx [hunk 1]
```diff
@@ -38,7 +32,9 @@ export type LocationDescriptor = Partial<Path> & {
  * </Button>
  *
  */
-export const Button = (props: ButtonProps) => {
+export const Button = <RecordType extends RaRecord = RaRecord>(
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/button/Button.tsx:35]: should default to any

File: packages/ra-ui-materialui/src/button/Button.tsx [hunk 2]
```diff
@@ -102,7 +98,7 @@ export const Button = (props: ButtonProps) => {
     );
 };
 
-interface Props {
+interface Props<RecordType extends RaRecord = RaRecord> {
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/button/Button.tsx:101]: Same, I think this is a breaking change

File: packages/ra-ui-materialui/src/button/Button.tsx (hunk 1 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/button/Button.tsx:35]: Actually, I failt to see why there is a `record` prop. This should be handled by the specialized buttons only imo

File: packages/ra-ui-materialui/src/button/Button.tsx [hunk 3]
```diff
@@ -114,13 +110,16 @@ interface Props {
     size?: 'small' | 'medium' | 'large';
     redirect?: RedirectionSideEffect;
     variant?: string;
-    // May be injected by Toolbar
-    record?: RaRecord;
+    // May be provided manually
+    record?: RecordType;
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/button/Button.tsx:114]: In fact, I fail to see what `record` does here. It's not even used in the code. This somehow seems to forbid usage of the generic `Button` outside of a `record` context, e.g. in a List, a dialog, a sidebar, or a custom page. I wonder why, as `Button` is purely a UI component.

**Comment by fzaninotto** [comment]
Could you please update the description to explain what problem this PR fixes?

**Pull request merged**